src/scripts/main.js

```diff
@@ -1,3 +1,428 @@
+  });
+
+  rowsArray.forEach((row) => tbody.appendChild(row));
+}
+
+function ascSortAge(index) {
+  const tbodyRows = document.querySelectorAll('tbody tr');
+  const rowsArray = Array.from(tbodyRows);
+
+  rowsArray.sort((a, b) => {
+    const ageA = Number(a.cells[index].innerHTML);
+    const ageB = Number(b.cells[index].innerHTML);
+
+    return ageA - ageB;
+  });
+
+  rowsArray.forEach((row) => tbody.appendChild(row));
+}
+
+function descSortAge(index) {
+  const tbodyRows = document.querySelectorAll('tbody tr');
+  const rowsArray = Array.from(tbodyRows);
+
+  rowsArray.sort((a, b) => {
+    const ageA = Number(a.cells[index].innerHTML);
+    const ageB = Number(b.cells[index].innerHTML);
+
+    return ageB - ageA;
+  });
+
+  rowsArray.forEach((row) => tbody.appendChild(row));
+}
+
+function ascSortSalary(index) {
+  const tbodyRows = document.querySelectorAll('tbody tr');
+  const rowsArray = Array.from(tbodyRows);
+
+  rowsArray.sort((a, b) => {
+    const srA = parseInt(a.cells[index].innerHTML.replace(/[$,]/g, ''), 10);
+    const srB = parseInt(b.cells[index].innerHTML.replace(/[$,]/g, ''), 10);
+
+    return srA - srB;
+  });
+
+  rowsArray.forEach((row) => tbody.appendChild(row));
+}
+
+function descSortSalary(index) {
+  const tbodyRows = document.querySelectorAll('tbody tr');
+  const rowsArray = Array.from(tbodyRows);
+
+  rowsArray.sort((a, b) => {
+    const srA = parseInt(a.cells[index].innerHTML.replace(/[$,]/g, ''), 10);
+    const srB = parseInt(b.cells[index].innerHTML.replace(/[$,]/g, ''), 10);
+
+    return srB - srA;
+  });
+
+  rowsArray.forEach((row) => tbody.appendChild(row));
+}
+
+thead.addEventListener('click', function (ev) {
+  let index;
+
+  if (ev.target.innerHTML === 'Name') {
+    index = 0;
+  }
+
+  if (ev.target.innerHTML === 'Position') {
+    index = 1;
+  }
+
+  if (ev.target.innerHTML === 'Office') {
+    index = 2;
+  }
+
+  if (ev.target.innerHTML === 'Age') {
+    index = 3;
+  }
+
+  if (ev.target.innerHTML === 'Salary') {
+    index = 4;
+  }
+
+  if (index === 3) {
+    if (ev.target.dataset.state === 'unsorted') {
+      ev.target.setAttribute('data-state', 'ascending');
+      ascSortAge(index);
+    } else if (ev.target.dataset.state === 'ascending') {
+      ev.target.setAttribute('data-state', 'descending');
+      descSortAge(index);
+    } else if (ev.target.dataset.state === 'descending') {
+      ev.target.setAttribute('data-state', 'ascending');
+      ascSortAge(index);
+    }
+  } else if (index === 4) {
+    if (ev.target.dataset.state === 'unsorted') {
+      ev.target.setAttribute('data-state', 'ascending');
+      ascSortSalary(index);
+    } else if (ev.target.dataset.state === 'ascending') {
+      ev.target.setAttribute('data-state', 'descending');
+      descSortSalary(index);
+    } else if (ev.target.dataset.state === 'descending') {
+      ev.target.setAttribute('data-state', 'ascending');
+      ascSortSalary(index);
+    }
+  } else {
```

Comment:
Consider will be better if you use switch case construction here and move some logic to the helper functions